cmd/agent/app/agent.go

```diff
@@ -229,7 +230,14 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 	restConfig := mgr.GetConfig()
 	dynamicClientSet := dynamic.NewForConfigOrDie(restConfig)
 	controlPlaneInformerManager := genericmanager.NewSingleClusterInformerManager(dynamicClientSet, 0, stopChan)
-	resourceInterpreter := resourceinterpreter.NewResourceInterpreter(controlPlaneInformerManager)
+	controlPlaneKubeClientSet := kubeclientset.NewForConfigOrDie(restConfig)
+
+	sharedFactory := informers.NewSharedInformerFactory(controlPlaneKubeClientSet, 0)
+	serviceLister := sharedFactory.Core().V1().Services().Lister()
+	sharedFactory.Start(stopChan)
+	sharedFactory.WaitForCacheSync(stopChan)
```

Comment:
Generally looks good to me. The only concern is that this(setting up the informer) may slow down the controller startup. 

It will be better to have a comment here to explain why we need the serviceLister.